libs/types/src/tokens.rs

```diff
@@ -41,6 +41,10 @@ pub enum CurrencyId {
 
 	/// A foreign asset
 	ForeignAsset(ForeignAssetId),
+
+	/// An emulated internal, non-transferrable currency
+	/// Its issuance and holding is handled inherently
+	BlockRewards,
```

Comment:
Why the change? I liked the id based approach. My reasoning here is that we will most likely need another currency CfgStake or so, that user can get by submitting CFG and then stake on pools or loans or so with it. I think it makes sense to have at least the following

```rust
pub enum CurrencyId {
     ....
     Staking(StakingCurrency) 
}

pub enum StakingCurrency {
    BlockRewards,
}
```

But just a thought out of my head. I also would like to get feedback on that. We can also go with the current approach, I just have the feeling that we are slowly cluttering the enum here and it gets really dirty. Just like `AUSD` is not really a good varaint, as we are not even having it nor have we a plan on how we want to extend the enum here with all the connectors tokens.

Reply:
I mistook your silence and @lemunozm's rocket emoji [here](https://github.com/centrifuge/centrifuge-chain/pull/1198#discussion_r1131109660) as approval to go with the proposal. I was under the assumption we are not planning to add more "staking" currency ids anytime soon. However, I totally agree with the cluttering aspect and prefer this option here.

Reply:
My rocket has been a destruction missile 😆. Currently, I'm not seeing another currency related to this (I mean, a special internal currency for reward purposes). But I'm ok with whatever solution we choose.